### DISCUSSION TITLE
WaterMark is not visible on the chart

### DISCUSSION THREAD

**Discussion by christose** [tradingview/lightweight-charts#680] [General]
Hello,

I am setting a WaterMark but is not visible on the chart, below is my code:

```
this.WaterMark = {
      color: obj.open < obj.close ? this.blue_indicator : this.red_indicator,
      visible: true,
      text: "O:" + obj.open + " H:" + obj.high + "  L:" + obj.low + "  C:" + obj.close + "",
      fontSize: 24,
      horzAlign: 'right',
      vertAlign: 'top',
    } as WatermarkOptions;
    this.ChartSeries.applyOptions(this.WaterMark);
```
Here's a screenshot with the chart where the watermark is not visible.
![image](https://user-images.githubusercontent.com/8969889/105161475-15537380-5b1a-11eb-83d8-294e5c4f67d2.png)
The above code is being called on CrosshairMove only once. (for testing purposes).

I assume that the WaterMark is the one in the screenshots below? (red arrow):
![image](https://user-images.githubusercontent.com/8969889/105160992-8ba3a600-5b19-11eb-96f1-bccd1030ac21.png)
![image](https://user-images.githubusercontent.com/8969889/105161022-93fbe100-5b19-11eb-9fdf-e3cce640dd94.png)

Assuming the watermark is the one I am pointing, can I achieve such a watermark like in the candle series in tradingview, with OHLC in one color and values in another color?

**Comment by RubberArchind** [comment]
@christose It's not a watermark, what you looking for is a legend you can see it [here](https://www.tradingview.com/lightweight-charts/)

**Reply by timocov** [reply]
That's right, what @christose is pointing to on the screenshots is legend, not watermark, and lightweight-charts doesn't provide legend widget out of the box. But you can implement it yourself though.

**Reply by christose** [reply]
Thank you guys, I appreciate it, I've done it and it works like a harm!

**Reply by fs10218** [reply]
Is it possible to margin the watermark?

**Comment by timocov** [comment]
> `this.ChartSeries.applyOptions(this.WaterMark);`

See our test case how to set up watermark - https://github.com/tradingview/lightweight-charts/blob/master/tests/e2e/graphics/test-cases/initial-options/watermark.js#L17:
- it's NOT part of the series
- watermark's options should be inside an object with a `watermark` key in chart's options (not series).

**Reply by christose** [reply]
Thanks I fixed it, my silly mistake sorry
